assets/eip-5216/ERC1155ApprovalByAmount.sol

```diff
@@ -108,7 +110,9 @@ abstract contract ERC1155ApprovalByAmount is ERC1155, IERC1155ApprovalByAmount {
 
         require(idsLength == amountsLength, "ERC1155ApprovalByAmount: ids and amounts length mismatch");
         for (uint256 i = 0; i < idsLength;) {
-            require(allowance(from, to, ids[i]) >= amounts[i], "ERC1155ApprovalByAmount: operator is not approved for that id or amount");
+            if(allowance(from, to, ids[i] < amounts[i])) {
```

Comment:
What's the rationale to not use `_allowances` instead of using `allowances()`?

Reply:
Mmm that's true, thanks for feedback. Also that `if` won't work fine haha. I'll fix and I'll include new commit!

Reply:
Done @xinbenlv . Thank you very much for your feedback!